### PR TITLE
Indices support

### PR DESCRIPTION
**Pull request by HunterL**
This PR addresses the following

- Update `rest.json` to include `indices` in `asset_class`
- Creating an `indicesClient` & attach to PolygonRestClient for the purpose of calls to Indices API
- Add snapshot endpoint to `indicesClient`
- Add previous close endpoint
- Add daily open/close endpoint
- Test that generalized aggregates endpoint works for indices tickers (e.g. "I:SPX")
- Update websockets for marshaling indices events

### CODE REVIEW

File: src/main/kotlin/io/polygon/kotlin/sdk/Version.kt
Comment (HunterL) [inline src/main/kotlin/io/polygon/kotlin/sdk/Version.kt:4]: Specific call out here that I'm bumping minor version since there is a new feature with no breaking changes. Let me know if v4.2.2` would be more appropriate.

**Review by mmoghaddam385 — APPROVED** [review]
LGTM, I didn't look too closely at the fields themselves so you might want to wait for another review from someone more familiar with indices models

File: src/main/kotlin/io/polygon/kotlin/sdk/rest/indices/PreviousClose.kt
Comment (antdjohns) [inline src/main/kotlin/io/polygon/kotlin/sdk/rest/indices/PreviousClose.kt:27]: no concept of adjusted in indices. Check other apis, I don't think this should be present.

Comment (antdjohns) [reply src/main/kotlin/io/polygon/kotlin/sdk/rest/indices/PreviousClose.kt:27]: unless this is covering more than just indices...

File: src/main/kotlin/io/polygon/kotlin/sdk/rest/indices/PolygonIndicesClient.kt
Comment (antdjohns) [reply src/main/kotlin/io/polygon/kotlin/sdk/rest/indices/PolygonIndicesClient.kt:74]: all "unadjusted" parameters should be removed, if they can.

**Pull request merged**
